command/agent/http.go

```diff
@@ -650,12 +652,18 @@ func (s *HTTPServer) handleUI(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		header := w.Header()
 		agentConfig := s.agent.GetConfig()
```

Comment:
Note that you've dropped the nil-check of `agentConfig.UI` here, so the new code will cause the agent to panic if the `ui` block is unset. Likewise for the `ContentSecurityPolicy` field.

You'll need a check for both nils, and at that point you might want to consider assigning to a variable like `policy := agentConfig.UI.ContentSecurityPolicy` just so you don't have to write out the chain of dereferences a half-dozen times. But see also my comment below about where this merge should live.

Reply:
The handleUI function is called after checking if the agentConfig.UI is not nil on line 509. That is why i did not recheck this.

Reply:
The ContentSecurityPolicy will never be nil because I added defaults values on the nomad/structs/config/ui.go file. IMHO